website_sale_custom_filter/static/src/js/website_sale_filters.js

```diff
@@ -0,0 +1,118 @@
+odoo.define("website_sale_custom_filter.price_range_option", function (require) {
+    "use strict";
+
+    const publicWidget = require("web.public.widget");
+
+    publicWidget.registry.CustomFilterCollapse = publicWidget.Widget.extend({
+        selector: "#wsale_products_custom_filters",
+        events: {
+            "click .fa-chevron-right": "_onOpenClick",
+            "click .fa-chevron-down": "_onCloseClick",
+        },
+
+        // --------------------------------------------------------------------------
+        // Handlers
+        // --------------------------------------------------------------------------
+
+        /**
+         * @private
+         * @param {Event} ev
+         */
+        _onOpenClick: function (ev) {
+            var $fa = $(ev.currentTarget);
+            $fa.parents("li").children()[1].style.display = "inline-block";
+            $fa.toggleClass("fa-chevron-down fa-chevron-right");
+        },
+        /**
+         * @private
+         * @param {Event} ev
+         */
+        _onCloseClick: function (ev) {
+            var $fa = $(ev.currentTarget);
+            $fa.parents("li").children()[1].style.display = "none";
+            $fa.toggleClass("fa-chevron-right fa-chevron-down");
+        },
+    });
+
+    publicWidget.registry.CheckboxCustomFilterSelector = publicWidget.Widget.extend({
+        selector: "#wsale_products_custom_filters",
+        events: {
+            'change input[type="checkbox"]': "_onChangeCustomAttribute",
+        },
+
+        /**
+         * @private
+         * @param {Event} ev
+         */
+        _onChangeCustomAttribute(ev) {
+            const search = $.deparam(window.location.search.substring(1));
```

Comment:
Not a blocker but you can use URLSearchParam both to parse and to build the url later